File: work/stellar-restomod/debug_receiver.js
const http = require('http');
const fs = require('fs');

const server = http.createServer((req, res) => {
    // Enable CORS
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'POST, GET, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

    if (req.method === 'OPTIONS') {
        res.writeHead(200);
        res.end();
        return;
    }

    if (req.method === 'POST' && req.url === '/log') {
        let body = '';
        req.on('data', chunk => {
            body += chunk.toString();
        });
        req.on('end', () => {
            fs.writeFileSync('debug_output.txt', body, 'utf8');
            console.log('Received debug data and wrote to debug_output.txt');
            res.writeHead(200, { 'Content-Type': 'application/json' });
            res.end(JSON.stringify({ status: 'ok' }));
        });
    } else {
        res.writeHead(404);
        res.end();
    }
});

server.listen(8089, 'localhost', () => {
    console.log('Debug receiver server listening on http://localhost:8089');
});

File: work/stellar-restomod/debug_receiver.py
import http.server
import socketserver
import json

class DebugHandler(http.server.BaseHTTPRequestHandler):
    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'POST, GET, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()

    def do_POST(self):
        if self.path == '/log':
            content_length = int(self.headers['Content-Length'])
            post_data = self.rfile.read(content_length).decode('utf-8')
            
            with open('debug_output.txt', 'w', encoding='utf-8') as f:
                f.write(post_data)
                
            self.send_response(200)
            self.send_header('Access-Control-Allow-Origin', '*')
            self.send_header('Content-Type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({'status': 'ok'}).encode('utf-8'))
            print("Received debug log and wrote to debug_output.txt")
        else:
            self.send_response(404)
            self.end_headers()

PORT = 8089
with socketserver.TCPServer(("127.0.0.1", PORT), DebugHandler) as httpd:
    print(f"Debug Python receiver listening on port {PORT}")
    httpd.serve_forever()
